File: bumperbot_cpp_examples/src/simple_publisher.cpp
#include <rclcpp/rclcpp.hpp>
#include <std_msgs/msg/string.hpp>
#include <chrono>

using namespace std::chrono_literals;

class simplePublisher : public rclcpp::Node
{
    public:
        simplePublisher() : Node("anand_pub"), counter_(0) //Constructor
        {
           pub_ =  create_publisher<std_msgs::msg::String>("anandcpp",10); //create publisher and assign to shared pointer
           timer_ = create_wall_timer(1s, std::bind(&simplePublisher::timer_callback,this)); //create timer callback 

           RCLCPP_INFO(get_logger(), "anand here doing c++");
        }

    private:
        unsigned int counter_ ;
        rclcpp::Publisher<std_msgs::msg::String>::SharedPtr pub_; //create a shared pointer of the given data type for publisher
        rclcpp::TimerBase::SharedPtr timer_;  //create a shared pointer for the timer

        void timer_callback(){
            auto message = std_msgs::msg::String();
            message.data = "Anand hu mai!" + std::to_string(counter_++);
            pub_->publish(message);    // publishing the data through shared pointer of publisher
        }
};

int main(int argc, char* argv[]){
    rclcpp::init(argc, argv);
    auto node = std::make_shared<simplePublisher>();  //create a shared pointer of class obj
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: bumperbot_py_examples/bumperbot_py_examples/simple_publisher.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class simplePublisher(Node):
    def __init__(self):
        super().__init__("simple_publisher") #name of node

        self.pub_ = self.create_publisher(String, "anandtopic",10)
        self.counter_ = 0
        self.frequency_ = 1.0

        self.get_logger().info("Hi Anand here %d" % self.frequency_)
        self.timer_ = self.create_timer(self.frequency_,self.timer_callback)

    def timer_callback(self):
        msg = String()
        msg.data = "Hello Anand %d" % self.counter_

        self.pub_.publish(msg)
        self.counter_ +=1

def main():
    rclpy.init()
    anand = simplePublisher()
    rclpy.spin(anand)
    anand.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()